flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java

```diff
@@ -145,7 +150,17 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
 		final FlinkKafkaProducer<RowData> kafkaProducer =
 				createKafkaProducer(keySerialization, valueSerialization);
 
-		return SinkFunctionProvider.of(kafkaProducer);
+		return new SinkFunctionProvider() {
```

Comment:
Could you add a new util method in `SinkFunctionProvider`? I think most of the connector can use this new method. 

```java
	/**
	 * Helper method for creating a SinkFunction provider with a provided sink parallelism.
	 */
	static SinkFunctionProvider of(SinkFunction<RowData> sinkFunction, Integer sinkParallelism) {
		return new SinkFunctionProvider() {

			@Override
			public SinkFunction<RowData> createSinkFunction() {
				return sinkFunction;
			}

			@Override
			public Optional<Integer> getParallelism() {
				return Optional.ofNullable(sinkParallelism);
			}
		};
	}
```

Reply:
yes,it's better to have common function